irohad/ametsuchi/impl/postgres_setting_query.hpp

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_POSTGRES_SETTING_QUERY_HPP
+#define IROHA_POSTGRES_SETTING_QUERY_HPP
+
+#include "ametsuchi/setting_query.hpp"
+
+#include <soci/soci.h>
+#include <boost/optional.hpp>
+#include "logger/logger_fwd.hpp"
+
+namespace iroha {
+  namespace ametsuchi {
+
+    /**
+     * Class which implements SettingQuery with a Postgres backend.
+     */
+    class PostgresSettingQuery : public SettingQuery {
+     public:
+      PostgresSettingQuery(std::unique_ptr<soci::session> sql,
+                           logger::LoggerPtr log);
+
+      expected::Result<
+          std::shared_ptr<const shared_model::validation::Settings>,
```

Comment:
minor issue: object producers should generally not return `shared_ptr` withuot a reason. when you create an object, it is usually not shared with anyone, and the user should decide the type of smart pointer to use. `unique_ptr` can be seamlessly converted to `shared_ptr`, but the reverse does not hold. also, `unique_ptr` are cheaper in operation. so I recommend
```suggestion
          std::unique_ptr<const shared_model::validation::Settings>,
```

Reply:
Fixed